views/js/previewer/component/qtiTest.js

```diff
@@ -81,6 +81,37 @@ define([
             const dataHolder = runner.getDataHolder();
             const testMap = dataHolder.get('testMap');
             const item = mapHelper.getItemAt(testMap, 0);
+            runner.on('nav-next', () => {
+                const newTextContext = dataHolder.get('testContext');
```

Comment:
Can you create function for the common stuff to avoid DRY?
Something like: 
```javascript
const loadItem = (position) => {
  const newTextContext = dataHolder.get('testContext');
  const newPosition = newTextContext.itemPosition + position;
  const newItem = mapHelper.getItemAt(testMap, newPosition);
      if (newItem && newItem.uri) {
        dataHolder.set('testContext', {
          itemIdentifier: testMap.jumps[newPosition].identifier,
          itemPosition: newPosition,
          testPartId: testMap.jumps[newPosition].part
        });
     return runner.loadItem(newItem.uri);
   } else {
     runner.trigger('finish leave');
    }
 };
                    
const loadNextItem = loadItem(1)
const loadPrevItem = loadItem(-1)
runner.on('nav-next', loadNextItem)
runner.on('nav-prev', loadPrevItem)
```

Reply:
I was thinking the same :)